src/core/aws-net-websocket-registry.adb

```diff
@@ -161,10 +161,14 @@ package body AWS.Net.WebSocket.Registry is
       entry Get_Socket (WebSocket : out Object_Class);
       --  Get a WebSocket having some data to be sent
 
-      procedure Release_Socket (WebSocket : Object_Class);
+      procedure Release_Socket (WebSocket : in out Object_Class);
       --  Release a socket retrieved with Get_Socket above, this socket will be
       --  then available again.
 
+      procedure Free (WebSocket : in out Object_Class);
+      --  Free WebSocket immediately if not taken by another tasK, otherwise
```

Comment:
"tasK" -> "task"

Reply:
There is a confusion in the API because the websocket itself has a primitive operation Free (aws-net-websocket.adb:215) which does an immediate deallocation. Shouldn't the latter internally call `Release_Socket` instead, and perform the actual deallocation later as needed ? As it is, the user might still free the socket directly, breaking the code. Perhaps you have an idea how to make this clearer ?
At the very least, it seems we should document that behavior, and perhaps rename AWS.Net.WebSocket.Registry.DB.Free to something like `Deferred_Free` or something to clear possible confusion in the AWS code itself ?

Reply:
I thought about the `Deferred_Free` naming, not bad but confusing in the other direction as the implementation does free immediately if the socket is not currently handled.

Reply:
`Free_Or_Defer` ?

Reply:
Works for me! I'll do the change. Thanks.